pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -164,9 +167,32 @@ public HelixServerStarter(String helixClusterName, String zkServer, Configuratio
     serverMetrics.addCallbackGauge("memory.allocationFailureCount", PinotDataBuffer::getAllocationFailureCount);
   }
 
-  private void updateInstanceConfigInHelix(int adminApiPort, boolean shuttingDown) {
+  private void waitForAllSegmentsLoaded() {
+    if (_helixServerConfig.getBoolean(CommonConstants.Server.CONFIG_OF_STARTER_ENABLE_SEGMENTS_LOADING_CHECK, CommonConstants.Server.DEFAULT_STARTER_ENABLE_SEGMENTS_LOADING_CHECK)) {
+      long startTime = System.currentTimeMillis();
+      int serverStarterTimeout = _helixServerConfig.getInt(CommonConstants.Server.CONFIG_OF_STARTER_TIMEOUT_IN_SECONDS, CommonConstants.Server.DEFAULT_STARTER_TIMEOUT_IN_SECONDS);
+      boolean allSegmentsLoaded = false;
+      while (System.currentTimeMillis() - startTime < 1000 * serverStarterTimeout) {
+        if (ServiceStatus.getServiceStatus() == Status.GOOD) {
+          LOGGER.info("All the segments are fully loaded into Pinot server.");
+          allSegmentsLoaded = true;
+          break;
+        }
+        try {
+          LOGGER.info("Waiting for all segments loaded, sleep 10 seconds...");
+          Thread.sleep(10000);
+        } catch (Exception e) {
+          // Do nothing
```

Comment:
Please catch interrupted exception separately. 
When you catch interrupted exception, do:
{
Thread.currentThread.interrupt();
return;
}
This allows the interrupt to propagate - the method should not continue on ignoring the interrupt.
For all other exceptions please log the exception.

Reply:
will do